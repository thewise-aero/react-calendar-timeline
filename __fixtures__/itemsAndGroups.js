
export const items = [
  {
    id: '0',
    group: '1',
    start_time: 1540540000000,
    end_time: 1540546803877,
    canMove: false,
    canResize: false,
    title: 'title 0',
  },
  {
    id: '5',
    group: '1',
    start_time: 1540532800000,
    end_time: 1540554003877,
    canMove: false,
    canResize: false,
    className: '',
    title: 'title 5',
  },
  {
    id: '6',
    group: '1',
    start_time: 1540550800000,
    end_time: 1540575603877,
    canMove: false,
    canResize: false,
    className: '',
    title: 'title 6',
  },
  {
    id: '1',
    group: '1',
    start_time: 1540570000000,
    end_time: 1540584875919,
    canMove: true,
    canResize: 'both',
    title: 'title 1',
  },
  {
    id: '2',
    group: '1',
    start_time: 1540620000000,
    end_time: 1540640397548,
    canMove: false,
    canResize: false,
    className: '',
    title: 'title 2',
  },
  {
    id: '3',
    group: '3',
    start_time: 1540656000000,
    end_time: 1540676397548,
    canMove: false,
    canResize: false,
    className: '',
    title: 'title 3',
  }
]

export const groups = [{ id: '1' }, { id: '2' }, { id: '3' }]
