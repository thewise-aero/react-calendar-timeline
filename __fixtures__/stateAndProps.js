import { defaultKeys } from 'lib/default-config'
import {items} from './itemsAndGroups'
import Timeline from 'lib/Timeline'

export const visibleTimeStart = 1540501200000
export const visibleTimeEnd = 1540587600000

export const props = {
  ...Timeline.defaultProps,
  keys: defaultKeys,
  lineHeight: 30,
  stackItems: true,
  itemHeightRatio: 0.75,
  visibleTimeEnd,
  visibleTimeStart,
}

export const propsNoStack = {
  ...props,
  stackItems: false,
}


export const state = {
  draggingItem: undefined,
  dragTime: null,
  resizingItem: null,
  resizingEdge: null,
  resizeTime: null,
  newGroupId: null,
  selectedItem: null,
  canvasTimeStart: 1540414800000,
  visibleTimeEnd: visibleTimeEnd,
  visibleTimeStart: visibleTimeStart,
  canvasTimeEnd: 1540674000000,
  width: 1000,
}

//offset 1 hour
const timeOffset = 1 * 60 *60 *1000

export const stateMoveItem = {
  ...state,
  draggingItem: items[0].id,
  dragTime: items[0].start_time+timeOffset,
  newGroupId: items[0].group,
}
export const stateResizeItemLeft = {
  ...state,
  resizingItem: items[0].id,
  resizingEdge: 'left',
  resizeTime: items[0].start_time+timeOffset,
  newGroupId: items[0].group,
}

export const stateResizeItemRight = {
  ...state,
  resizingItem: items[0].id,
  resizingEdge: 'right',
  resizeTime: items[0].end_time+timeOffset,
  newGroupId: items[0].group,
}