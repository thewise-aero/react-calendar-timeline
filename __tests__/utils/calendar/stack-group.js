import { stackGroup } from 'lib/utility/calendar'
import { dimensionItems } from '../../../__fixtures__/groupOrderAndItemDimentions'

describe('stackGroup', ()=>{
    it('should stack list of items', ()=>{
        expect(stackGroup(dimensionItems, true, 30)).toMatchSnapshot()
    })
    it('should not stack list of items', ()=>{
        expect(stackGroup(dimensionItems, false, 30)).toMatchSnapshot()
    })
})