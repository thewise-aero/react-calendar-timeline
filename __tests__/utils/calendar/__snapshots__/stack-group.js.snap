// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`stackGroup should not stack list of items 1`] = `
Object {
  "groupHeight": 30,
  "verticalMargin": 0,
}
`;

exports[`stackGroup should stack list of items 1`] = `
Object {
  "groupHeight": 30,
  "verticalMargin": 3.75,
}
`;
